patches/minecraft/net/minecraft/entity/passive/EntityHorse.java.patch

```diff
@@ -0,0 +1,63 @@
+--- ../src-base/minecraft/net/minecraft/entity/passive/EntityHorse.java
++++ ../src-work/minecraft/net/minecraft/entity/passive/EntityHorse.java
+@@ -34,6 +34,7 @@
+     private static final UUID field_184786_bD = UUID.fromString("556E1665-8B10-40C8-8F9D-CF9B1667F295");
+     private static final DataParameter<Integer> field_184789_bG = EntityDataManager.<Integer>func_187226_a(EntityHorse.class, DataSerializers.field_187192_b);
+     private static final DataParameter<Integer> field_184791_bI = EntityDataManager.<Integer>func_187226_a(EntityHorse.class, DataSerializers.field_187192_b);
++    private static final DataParameter<String> HORSE_ARMOR_STRING = EntityDataManager.<String>func_187226_a(EntityHorse.class, DataSerializers.field_187194_d);
```

Comment:
It's string based because that prevents load order changes causing strange behaviour. The vanilla system is ordinal based. Mod removal, addition or dependency declaration would all cause the ordinals of each enum constant to change, which would result in the wrong armour appearing on a horse. I could add an ID field to HorseArmorType, but then ID collisions are more likely.